upki/src/config.rs

```diff
@@ -0,0 +1,69 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use eyre::{Context, OptionExt, Report};
+use serde::{Deserialize, Serialize};
+
+/// `upki` configuration.
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Config {
+    /// Configuration for crlite-style revocation.
+    pub revocation: Revocation,
+}
+
+impl Config {
+    /// Load the configuration data from a file at `path`.
+    pub fn load_from_file(path: &Path) -> Result<Self, Report> {
+        let config_content = fs::read_to_string(path)?;
+        toml::from_str(&config_content).wrap_err("cannot parse configuration file")
+    }
+
+    /// Look for an existing configuration file.
+    ///
+    /// Returns:
+    /// - `Ok(Some(path))` giving the found configuration file path.
+    /// - `Ok(None)` if no such configuration file exists.
+    /// - `Err(...)` if we couldn't determine if such a file exists (eg, `$HOME` is not set, or another XDG
+    ///   environment variable is malformed)
+    pub fn find_config_file() -> Result<Option<PathBuf>, Report> {
+        Ok(xdg::BaseDirectories::with_prefix(PREFIX).find_config_file(CONFIG_FILE))
```

Comment:
Nit: import `BaseDirectories`? Could this document the conceptual/expected search strategy? I'm somewhat inclined to make this a free function (like `config::find_path()`?).